cleverhans/experimental/certification/optimization.py

```diff
@@ -282,6 +326,20 @@ def run_one_step(self, eig_init_vec_val, eig_num_iter_val, smooth_val,
                                 str(self.current_step) + '.json')
         with tf.gfile.Open(filename) as file_f:
           file_f.write(stats)
+
+    # Project onto feasible set of dual variables
+    if self.current_step % self.params['projection_steps'] == 0 and self.current_unconstrained_objective < 0:
+      nu = self.sess.run(self.dual_object.nu)
+      feed_dict = {
```

Comment:
You first compute `self.dual_object.get_lanczos_eig`, but at the same time you populate feed_dict with a bunch of stuff from `projected_dual_object`. I would suggest you to prepare two feed_dicts, one for `self.dual_object.get_lanczos_eig` which only refers to `self.dual_object` and then another one for `self.projected_dual_object.compute_certificate` which would only refer to `self.projected_dual_object`